File: docs/samples/github_api.py
# Real-world schema: GitHub API types (Pydantic v2)
from pydantic import BaseModel, Field
from typing import Optional, List
from enum import Enum


class UserType(str, Enum):
    user = "User"
    organization = "Organization"
    bot = "Bot"


class User(BaseModel):
    id: int
    login: str
    avatar_url: str
    type: UserType = Field(alias="type")
    site_admin: bool


class Label(BaseModel):
    id: int
    name: str
    color: str
    description: Optional[str] = None


class MilestoneState(str, Enum):
    open = "open"
    closed = "closed"


class Milestone(BaseModel):
    id: int
    number: int
    title: str
    description: Optional[str] = None
    state: MilestoneState
    open_issues: int
    closed_issues: int
    due_on: Optional[str] = None


class IssueState(str, Enum):
    open = "open"
    closed = "closed"


class Issue(BaseModel):
    id: int
    number: int
    title: str
    body: Optional[str] = None
    state: IssueState
    user: User
    labels: List[Label]
    assignees: List[User]
    milestone: Optional[Milestone] = None
    comments: int
    closed_at: Optional[str] = None


class Repository(BaseModel):
    id: int
    name: str
    full_name: str
    description: Optional[str] = None
    private: bool
    owner: User
    html_url: str
    clone_url: str
    default_branch: str
    language: Optional[str] = None
    stargazers_count: int
    forks_count: int
    open_issues_count: int
    topics: List[str]
    created_at: str
    updated_at: str
    pushed_at: Optional[str] = None


class GitRef(BaseModel):
    label: str
    ref: str = Field(alias="ref")
    sha: str
    user: User
    repo: Repository


class PullRequestState(str, Enum):
    open = "open"
    closed = "closed"


class PullRequest(BaseModel):
    id: int
    number: int
    title: str
    body: Optional[str] = None
    state: PullRequestState
    user: User
    head: GitRef
    base: GitRef
    merged: bool
    mergeable: Optional[bool] = None
    merged_by: Optional[User] = None
    comments: int
    commits: int
    additions: int
    deletions: int
    changed_files: int
    labels: List[Label]
    requested_reviewers: List[User]

File: docs/samples/github_api.rs
// Real-world schema: GitHub API types (simplified)
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Repository {
    pub id: i64,
    pub name: String,
    pub full_name: String,
    pub description: Option<String>,
    pub private: bool,
    pub owner: User,
    pub html_url: String,
    pub clone_url: String,
    pub default_branch: String,
    pub language: Option<String>,
    pub stargazers_count: i64,
    pub forks_count: i64,
    pub open_issues_count: i64,
    pub topics: Vec<String>,
    pub created_at: String,
    pub updated_at: String,
    pub pushed_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct User {
    pub id: i64,
    pub login: String,
    pub avatar_url: String,
    #[serde(rename = "type")]
    pub user_type: UserType,
    pub site_admin: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "PascalCase")]
pub enum UserType {
    User,
    Organization,
    Bot,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PullRequest {
    pub id: i64,
    pub number: i64,
    pub title: String,
    pub body: Option<String>,
    pub state: PullRequestState,
    pub user: User,
    pub head: GitRef,
    pub base: GitRef,
    pub merged: bool,
    pub mergeable: Option<bool>,
    pub merged_by: Option<User>,
    pub comments: i64,
    pub commits: i64,
    pub additions: i64,
    pub deletions: i64,
    pub changed_files: i64,
    pub labels: Vec<Label>,
    pub requested_reviewers: Vec<User>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum PullRequestState {
    Open,
    Closed,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GitRef {
    pub label: String,
    #[serde(rename = "ref")]
    pub ref_name: String,
    pub sha: String,
    pub user: User,
    pub repo: Repository,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Label {
    pub id: i64,
    pub name: String,
    pub color: String,
    pub description: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Issue {
    pub id: i64,
    pub number: i64,
    pub title: String,
    pub body: Option<String>,
    pub state: IssueState,
    pub user: User,
    pub labels: Vec<Label>,
    pub assignees: Vec<User>,
    pub milestone: Option<Milestone>,
    pub comments: i64,
    pub closed_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum IssueState {
    Open,
    Closed,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Milestone {
    pub id: i64,
    pub number: i64,
    pub title: String,
    pub description: Option<String>,
    pub state: MilestoneState,
    pub open_issues: i64,
    pub closed_issues: i64,
    pub due_on: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum MilestoneState {
    Open,
    Closed,
}
